src/scripts/main.js

```diff
@@ -1,3 +1,222 @@
+
+      const checkNumberCellA = numberFormat(cellA);
+      const checkNumberCellB = numberFormat(cellB);
+
+      if (Number(checkNumberCellA)) {
+        return ASC
+          ? checkNumberCellA - checkNumberCellB
+          : checkNumberCellB - checkNumberCellA;
+      }
+
+      return ASC ? cellA.localeCompare(cellB) : cellB.localeCompare(cellA);
+    });
+
+    rows.forEach((row) => tbody.appendChild(row));
+  });
+});
+
+const formFields = {
+  name: {
+    label: 'Name',
+    name: 'name',
+    type: 'text',
+    'data-qa': 'name',
+    validation: {
+      minLength: 4,
+    },
+  },
+  position: {
+    label: 'Position',
+    name: 'position',
+    type: 'text',
+    'data-qa': 'position',
+  },
+  office: {
+    label: 'Office',
+    name: 'office',
+    'data-qa': 'office',
+    options: [
+      'Tokyo',
+      'Singapore',
+      'London',
+      'New York',
+      'Edinburgh',
+      'San Francisco',
+    ],
+  },
+  age: {
+    label: 'Age',
+    name: 'age',
+    type: 'number',
+    'data-qa': 'age',
+    validation: {
+      min: 18,
+      max: 90,
+    },
+  },
+  salary: {
+    label: 'Salary',
+    name: 'salary',
+    type: 'number',
+    'data-qa': 'salary',
+  },
+  startDate: '',
+};
+
+const numberFormat = (n) => Number(n.replace(/[^0-9.-]+/g, ''));
+
+// Add new employee form
+
+const createForm = (fields) => {
+  const formEl = document.createElement('form');
+
+  formEl.classList.add('new-employee-form');
+
+  for (const key in fields) {
+    if (fields[key]) {
+      const label = document.createElement('label');
+
+      label.textContent = `${fields[key].label}: `;
+
+      let field = document.createElement('input');
+
+      if (fields[key].hasOwnProperty('options')) {
+        field = document.createElement('select');
+
+        for (const option of fields[key].options) {
+          const optionEl = document.createElement('option');
+
+          optionEl.value = option;
+          optionEl.textContent = option;
+          field.appendChild(optionEl);
+        }
+      }
+
+      label.appendChild(field);
+
+      field.setAttribute('name', fields[key].name);
+      field.setAttribute('type', fields[key].type);
+      field.setAttribute('data-qa', fields[key]['data-qa']);
+      field.setAttribute('min', fields[key].validation?.min);
+      field.setAttribute('max', fields[key].validation?.max);
+      field.setAttribute('minlength', fields[key].validation?.minLength);
+      field.setAttribute('required', true);
+      formEl.appendChild(label);
+    }
+  }
+
+  return formEl;
+};
+
+const showNotification = (type, message) => {
+  const notification = document.createElement('div');
+
+  notification.innerHTML = `<h2 class='title'>${message}</h2>`;
+
+  notification.classList.add('notification', type);
+  notification.setAttribute('data-qa', 'notification');
+
+  body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.remove();
+  }, 5000);
+};
```

Comment:
This function creates a new notification every time it is called and then removes it after 5 seconds. This could lead to performance issues if the function is called multiple times in a short period. Consider using a single notification element and updating its content and visibility as needed.